va/va_enc_av1.h

```diff
@@ -0,0 +1,974 @@
+         */
+        uint32_t search_idx3 : 3;
+        /**
+         * \brief Value used as index into ref_frame_idx[] to indicate that frame
+         * will be included in the reference list.
+         * valid value range: [1..7], invalid when value is 0.
+         */
+        uint32_t search_idx4 : 3;
+        /**
+         * \brief Value used as index into ref_frame_idx[] to indicate that frame
+         * will be included in the reference list.
+         * valid value range: [1..7], invalid when value is 0.
+         */
+        uint32_t search_idx5 : 3;
+        /**
+         * \brief Value used as index into ref_frame_idx[] to indicate that frame
+         * will be included in the reference list.
+         * valid value range: [1..7], invalid when value is 0.
+         */
+        uint32_t search_idx6 : 3;
+
+        /** \brief Reserved bytes for future use, must be zero. */
+        uint32_t Reserved    : 11;
+	} fields;
+	uint32_t value;
+} VARefFrameCtrlAV1;
+
+/**
+ * \brief AV1 Encoding Picture Parameter Buffer Structure.
+ *
+ * This structure conveys picture level parameters.
+ *
+ */
+typedef struct  _VAEncPictureParameterBufferAV1
+{
+    /** \brief AV1 encoder may support SupRes and dynamic scaling function.
+     *  For SupRes, underline encoder is responsible to do downscaling.
+     *  For dynamic scaling, app should provide the scaled raw source.
+     */
+    /** \brief Raw source frame width in pixels. */
+    uint16_t    frame_width_minus_1;
+    /** \brief Raw source frame height in pixels. */
+    uint16_t    frame_height_minus_1;
+
+    /** \brief Surface to store reconstructed frame, not used for enc only case. */
+    VASurfaceID reconstructed_frame;
+
+    /** \brief Buffer to store coded data. */
+    VABufferID  coded_buf;
+
+    /** \brief Reference frame buffers.
+     *  Each entry of the array specifies the surface index of the picture
+     *  that is referred by current picture or will be referred by any future
+     *  picture. The valid entries take value from 0 to 127, inclusive.
+     *  Non-valid entries, those do not point to pictures which are referred
+     *  by current picture or future pictures, should take value 0xFF.
+     *  Other values are not allowed.
+     *
+     *  Application should update this array based on the refreshing
+     *  information expected.
+     */
+    VASurfaceID reference_frames[8];
+
+    /** \brief Reference index list.
+     *  Contains a list of indices into refernce_frames[].
+     *  Indice with refernce frames range: [LAST_FRAME - LAST_FRAME,
+     *  LAST2_FRAME - LAST_FRAME, ..., ALTREF2_FRAME - LAST_FRAME].
+     *  #define LAST_FRAME 1
+     *  #define LAST2_FRAME 2
+     *  #define LAST3_FRAME 3
+     *  #define GOLDEN_FRAME 4
+     *  #define BWDREF_FRAME 5
+     *  #define ALTREF_FRAME 6
+     *  #define ALTREF2_FRAME 7
+     *  value range [0..7].
+     */
+    uint8_t     ref_frame_idx[7];
+
+    uint8_t     reserved8bits0;
```

Comment:
suppose it should be moved to behind order_hint, and it should be 3 bytes

Reply:
thx, move location and use 3 bytes.